cylc/flow/config.py

```diff
@@ -1531,24 +1530,37 @@ def configure_workflow_state_polling_tasks(self):
                 continue
             rtc = tdef.rtconfig
             comstr = (
-                "cylc workflow-state"
-                f" --task={tdef.workflow_polling_cfg['task']}"
-                " --point=$CYLC_TASK_CYCLE_POINT"
+                "cylc workflow-state "
+                f"{tdef.workflow_polling_cfg['workflow']}//"
+                "$CYLC_TASK_CYCLE_POINT/"
+                f"{tdef.workflow_polling_cfg['task']}"
             )
+            graph_trigger = tdef.workflow_polling_cfg['status']
+            config_trigger = rtc['workflow state polling']['output']
+            if (
+                graph_trigger is not None and
+                config_trigger is not None and
+                graph_trigger != config_trigger
+            ):
+                raise WorkflowConfigError(
+                    f'Polling task "{name}" must configure a target status or'
+                    f' output\nin the graph (:{graph_trigger}) or in its task'
+                    f' definition (output = "{config_trigger}") but not both.'
+                )
+            elif graph_trigger is not None:
+                comstr += f":{graph_trigger}"
+            elif config_trigger is not None:
+                comstr += f":{config_trigger}"
+            else:
+                # default to :succeeded
+                comstr += f":{TASK_OUTPUT_SUCCEEDED}"
+
             for key, fmt in [
-                    ('user', ' --%s=%s'),
-                    ('host', ' --%s=%s'),
                     ('interval', ' --%s=%d'),
                     ('max-polls', ' --%s=%s'),
-                    ('run-dir', ' --%s=%s')]:
+                    ('alt-cylc-run-dir', ' --%s=%s')]:
                 if rtc['workflow state polling'][key]:
                     comstr += fmt % (key, rtc['workflow state polling'][key])
-            if rtc['workflow state polling']['message']:
-                comstr += " --message='%s'" % (
-                    rtc['workflow state polling']['message'])
-            else:
-                comstr += " --status=" + tdef.workflow_polling_cfg['status']
-            comstr += " " + tdef.workflow_polling_cfg['workflow']
             script = "echo " + comstr + "\n" + comstr
```

Comment:
Alternative, unimportant
```suggestion
            script = "set +x\n" + comstr
```

Reply:
I think you mean `set -x`. But echo is simpler here. It's just intended to provide a record in the job.out of the command that was automatically put in the job script. Using `set -x` writes to stderr, which might make it appear that something went wrong, and it requires turning off again to avoid reporting the job script boiler-plate code as well.